### PR TITLE
More Django advice

### PR DESCRIPTION
**Pull request by hjwp**

### CODE REVIEW

**Comment by bobthemighty** [comment]
LGTM

**Comment by hjwp** [comment]
TODO: david suggests fat models are not so easy in practice, maybe omit or tone this down.

File: appendix_django.asciidoc [hunk 1]
```diff
@@ -366,8 +366,31 @@ that, and when might you still choose Django?
...
+
+* Repository and Unit of Work pattern are going to be quite a lot of work. The
+  main thing they will buy you in the short term is faster unit tests, so
+  evaluate whether that feels worth it in your case.  In the longer term, they
+  decouple your app from Django and the database, so if you anticipate wanting
+  to migrate away from either of those, Repository and UoW are a good idea.
+
+* You can still do DDD and domain modelling with Django models, tightly coupled
+  as they are to the database; you may be slowed down by migrations, but
+  it shouldn't be fatal.  So follow the "fat models" pattern, push as much
```
Comment (hjwp) [inline appendix_django.asciidoc]:
```suggestion
  it shouldn't be fatal.  So, as long as your apps is not too complex and your tests not too slow, you may
  be able to get something out of the "fat models" approach: push as much
```

**Comment by hjwp** [comment]
@seddonym i tried to tone down the fat models advice but i'm worried you may still think it endorses them too much? what do you think?

**Comment by seddonym** [comment]
> @seddonym i tried to tone down the fat models advice but i'm worried you may still think it endorses them too much? what do you think?

It's a bit better, but it still feels like an endorsement. Maybe the main point to drive home is to push the business logic away from views or forms. A common pattern is to put this on the models themselves (known as 'fat models'), or readers could try to keep their models focused on persistence logic, and locate business logic in dedicated modules.

We moved from fat models to a separate domain layer (which knows about models and has a lot of functions that accept models as arguments). It does help separate concerns to some extent, but database-independent testing is still difficult.

The other, perhaps obvious, point to make is that there's nothing to stop people factoring code so some of it is free of side effects. The fact that there are Django models somewhere doesn't mean you can't write code elsewhere that doesn't interact at all with models and is nice to test.

**Comment by hjwp** [comment]
merging this now, i think i got it to a reasonably sane place.

**Pull request merged**